### PR TITLE
Create Mock and Test for IdentifiedCase

### PR DESCRIPTION
**Pull request by Ion98**
Create a Mock and Test for IdentifiedCase

### CODE REVIEW

File: intellst-api/tests/Services/IdentifiedCaseHandlerTest.php [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        $dto->datePhoto = new DateTime();
+        $dto->firstDate = new DateTime();
+
+        return $dto;
+    }
+
+    /**
+     * @return AllowEntranceDTO
+     */
+    private function getAllowEntranceDTO(): AllowEntranceDTO
```
Comment (andreikojusko) [inline intellst-api/tests/Services/IdentifiedCaseHandlerTest.php:44]: method not used

File: intellst-api/tests/Services/IdentifiedCaseHandlerTest.php [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    {
+        self::bootKernel();
+
+        parent::setUp();
+
+        $this->em = static::$container->get('doctrine')->getManager();
+        $this->em->getConnection()->beginTransaction();
+    }
+
+    protected function tearDown(): void
```
Comment (andreikojusko) [inline intellst-api/tests/Services/IdentifiedCaseHandlerTest.php]: method not needed

File: intellst-api/tests/Services/IdentifiedCaseHandlerTest.php [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+{
+    private $em;
+
+    public function setUp()
+    {
+        self::bootKernel();
+
+        parent::setUp();
+
+        $this->em = static::$container->get('doctrine')->getManager();
```
Comment (andreikojusko) [inline intellst-api/tests/Services/IdentifiedCaseHandlerTest.php]: EntityManager not needed in this test

File: intellst-api/tests/Services/IdentifiedCaseHandlerTest.php [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+use App\DTO\AllowEntranceDTO;
+use App\DTO\IdentifiedCaseDTO;
+use App\Services\IdentifiedCaseHandler;
+use App\Transformer\IdentifiedCaseTransformer;
+use DateTime;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+use Doctrine\Common\Persistence\ObjectRepository;
+use Doctrine\ORM\EntityManagerInterface;
+
+class IdentifiedCaseHandlerTest extends KernelTestCase
```
Comment (andreikojusko) [inline intellst-api/tests/Services/IdentifiedCaseHandlerTest.php]: Kernel not needed, extend TestCase

File: intellst-api/tests/Services/IdentifiedCaseHandlerTest.php [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    private function getHandler(): IdentifiedCaseHandler
+    {
+        $repositoryMock = $this->createMock(ObjectRepository::class);
+        $emMock = $this->createMock(EntityManagerInterface::class);
+        $emMock
+            ->method('getRepository')
+            ->willReturn($repositoryMock);
+
+        return new IdentifiedCaseHandler($emMock,
+            static::$container->get('validator'),
```
Comment (andreikojusko) [inline intellst-api/tests/Services/IdentifiedCaseHandlerTest.php]: Use validator mock or instantiate a real validator

File: intellst-api/tests/Services/IdentifiedCaseHandlerTest.php [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+            static::$container->get(IdentifiedCaseTransformer::class));
```
Comment (andreikojusko) [inline intellst-api/tests/Services/IdentifiedCaseHandlerTest.php]: Instantiate real transformer

File: intellst-api/tests/Services/IdentifiedCaseHandlerTest.php [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+    }
+
+    /**
+     * @return IdentifiedCaseDTO
```
Comment (andreikojusko) [inline intellst-api/tests/Services/IdentifiedCaseHandlerTest.php]: not needed

File: intellst-api/tests/Services/IdentifiedCaseHandlerTest.php [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+    {
+        $handler = $this->getHandler();
+        $dto = $this->getIdentifiedCaseDTO();
+        $dto->photoFilename = '';
+        $result = $handler->updateIdentifiedCase($dto);
+
+        $this->assertCount(2, $result);
+        $this->assertEquals('photoFilename', $result->get(0)->getPropertyPath());
+        $this->assertEquals('This value should not be blank.', $result->get(0)->getMessage());
+    }
```
Comment (andreikojusko) [inline intellst-api/tests/Services/IdentifiedCaseHandlerTest.php:62]: please run the test with coverage and add more tests to cover more lines

**Review by andreikojusko — CHANGES_REQUESTED** [review]

File: intellst-api/tests/Services/IdentifiedCaseHandlerTest.php (hunk 1 above)
Comment (Ion98) [reply intellst-api/tests/Services/IdentifiedCaseHandlerTest.php:44]: > method not used

the method will be used by the person who will create the unit tests

File: intellst-api/tests/Services/IdentifiedCaseHandlerTest.php (hunk 8 above)
Comment (Ion98) [reply intellst-api/tests/Services/IdentifiedCaseHandlerTest.php:62]: > please run the test with coverage and add more tests to cover more lines

more tests will be done by other people

**Review by andreikojusko — APPROVED** [review]

**Pull request merged**